File: python/gsc/conf/route/rule.py
from typing import Optional

import pydantic


class Rule(pydantic.BaseModel):
    auth_user: Optional[list[str]] = None
    protocol: Optional[list[str]] = None
    domain_suffix: Optional[list[str]] = None
    ip_is_private: Optional[bool] = None
    clash_mode: Optional[str] = None
    rule_set: Optional[list[str]] = None
    outbound: str

File: src/internal/singbox/conf/route/rule.rs
use serde::{Deserialize, Serialize};

#[serde_with::skip_serializing_none]
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
pub struct Rule {
    pub auth_user: Option<Vec<String>>,
    pub protocol: Option<Vec<String>>,
    pub ip_is_private: Option<bool>,
    pub clash_mode: Option<String>,
    pub rule_set: Option<Vec<String>>,
    pub outbound: String,
}
